falcon/response_helpers.py

```diff
@@ -60,3 +61,18 @@ def format_range(value):
             str(value[0]) + '-' +
             str(value[1]) + '/' +
             str(value[2]))
+
+
+def is_ascii_encodable(s):  # pragma: no cover
+    """ check if argument encodes to ascii without error
+    """
+    if isinstance(s, six.text_type):
```

Comment:
Depends on what we want to protect against.
If you do the `str(s)` and `s` is a `str` created by say `s = "æøå"`, `str(s)` will do nothing. If you call `encode("ascii")` you will (correctly) get a `UnicodeDecodeError` (which most python2.7 folks working with non-ascii characters will be very familiar with).

I also did a tiny test of `is_ascii_encodable` with `str(s)` vs `s.encode("ascii")` and saw no difference.
if `str(s)` is faster, it is probably drowning in all the other things going on in that function.

@kgriffs Should I see a significant difference in performance between the two?
If so, how do we handle the case where the dev has an invalid `str` type?

if we removed the type checks, we could reduce the function to

```python
def is_ascii_encodable(s):  # pragma: no cover
    """Check if argument encodes to ascii without error."""
    try:
        s.encode("ascii")
    except UnicodeEncodeError:
        return False
    return True
```

Or

```python
def is_ascii_encodable(s):  # pragma: no cover
    """Check if argument encodes to ascii without error."""
    try:
        s.encode("ascii")
    except UnicodeEncodeError:
        return False
    except UnicodeDecodeError:
        return False
    return True
```

if we want to return False in the case that a py2 `str` contains non-ascii chars.

We could even catch `AttributeError` to handle the case where you accidentally hand it a NoneType, Dict or whatever.

If we use
```python
def is_ascii_encodable(s):  # pragma: no cover
    """Check if argument encodes to ascii without error."""
    try:
        s.encode("ascii")
    except UnicodeEncodeError:
        return False
    except UnicodeDecodeError:
        return False
    except AttributeError:
        return False
    return True
```

It will handle all sane uses for this function. You can still get a false-positive if you pass it  the _type_ `str`, (both py2 and py3), and i guess we could check for that.
_or_ we can accept that bad input returns bad output.

I think the last version is prob. the way to go.
We handle all valid input cases (str and unicode in py2 and str in py3) without doing tons of checking
to ensure the input is what we expected.

It _is_ a response helper, and not something that is supposed to face users.